Cargo.toml

```diff
@@ -14,18 +14,25 @@ description = "Fuel cryptographic primitives."
 borrown = "0.1"
 fuel-types = { version = "0.5", default-features = false }
 lazy_static = { version = "1.4", optional = true }
+serde = { version = "1.0", default-features = false, features = ["derive"], optional = true }
+sha2 = { version = "0.9", default-features = false }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
```

Comment:
Add a comment explaining why this is needed downstream

Reply:
Instead of checking target arch, why don't we include a feature `wasm` and require `secp256k1/rand` when it is enabled?

Using features is more flexible and powerful solution than if/else in the toml file